src/engine/strat_engine/blockdev.rs

```diff
@@ -55,10 +56,67 @@ pub fn resolve_devices(paths: &[&Path]) -> io::Result<BTreeSet<Device>> {
     Ok(devices)
 }
 
+#[derive(Debug)]
+pub struct BlockDevs {
```

Comment:
I see how we could pull out BlockDevs struct & impl into a separate module, but I don't see how we could avoid the struct + impl altogether. Explain a little more?

I also could see how `{read,write}_metadata()` could be `StratPool` methods...I think creating `BlockDevs` was in anticipation of `StratPool` getting really big and wanting to break it up (this is what happened with Froyo), but since this isn't the case yet, it's probably premature. I'll just put these in `StratPool` and drop `BlockDevs`.